### ISSUE TITLE
TownDeleteEvent does not pass Town object.

### ISSUE THREAD

**Issue by Mancye** [TownyAdvanced/Towny#2998]
The TownDeleteEvent is fired after the town is actually deleted so you can't use the Town object being deleted.

### What is your Towny version?
0.92.0

### What is your spigot/craftbukkit version?
1.13

**Comment by LlmDl** [comment]
Took a look at this after waking up, there's an event.getTownname(); which returns the string of the name of the town that was deleted. You could probably use this in your town-tracking plugin. Unless you're tracking the towns by mayor and not the town itself.

**Comment by LlmDl** [comment]
This has been solved as of 0.92.0.13 using the new PreDeleteTownEvent which can pass either townname or town object.

```
    - New Event: PreDeleteTownEvent
      - Passes either the Townname or Town object.
      - Useful for plugins developers who are storing town objects in their plugins and need to monitor deleted towns.
        - The normal DeleteTownEvent does not take town objects because by the time the event fires there is no town left.
```
